### PR TITLE
pastebinit: change upstream to maintained fork, add tests, and manpage

### PR DESCRIPTION
**Pull request by Samuel-Martineau**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by raboof** [comment]
Result of `nixpkgs-review pr 270974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pastebinit</li>
  </ul>
</details>

File: pkgs/tools/misc/pastebinit/test.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ runCommand
+, pastebinit
+, cacert
+, wget
+}:
+runCommand "pastebinit-test" { nativeBuildInputs = [ pastebinit cacert wget ]; } ''
+  export USER="nixpkgs-pastebinit-test"
+  teststring="test from pastebinit"
+  for pastebin in $(pastebinit -l | sed 1d | cut -d' ' -f2); do
+    echo "Trying $pastebin"
```
Comment (raboof) [inline pkgs/tools/misc/pastebinit/test.nix:10]: Hmm, did this work for you? I'm getting:
```
$ nix-build . -A pastebinit.tests
(...)
Trying dpaste.com
Failed to contact the server: <urlopen error [Errno -3] Temporary failure in name resolution>
error: builder for '/nix/store/gilyr1i9jz31mr6y8lx48vgmgvshlwaj-pastebinit-test.drv' failed with exit code 1;
```
As much as I like automated tests, perhaps we should skip the integration tests for now. Perhaps create a local end-to-end test as soon as someone packages a pastebin server for nixos? :)

**Comment by Samuel-Martineau** [comment]
I think I agree with you. Failing tests most likely indicate an error with the pastebin server more than the pastebin client itself, I think. I remember getting this error randomly, just waiting a bit, and then the tests randomly passing again. And this is without counting exceptions, like Debian's pastebin needing to be at least 3 lines long, or Ubuntu’s needing to be signed in to view (but ironically not to post). I’ll make another commit to remove the tests.

**Comment by raboof** [comment]
Result of `nixpkgs-review pr 270974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pastebinit</li>
  </ul>
</details>

File: pkgs/tools/misc/pastebinit/default.nix [hunk 2]
```diff
@@ -1,51 +1,55 @@
...
+    cp -a utils/* $out/bin
     cp -a pastebin.d $out/etc
-    substituteInPlace $out/bin/pastebinit --replace "'/etc/pastebin.d" "'$out/etc/pastebin.d"
+    substituteInPlace $out/bin/pastebinit --subst-var-by "etc" "$out/etc"
+    installManPage pastebinit.1
   '';
 
   meta = with lib; {
-    homepage = "https://launchpad.net/pastebinit";
+    homepage = "https://stgraber.org/category/pastebinit/";
```
Comment (raboof) [inline pkgs/tools/misc/pastebinit/default.nix:49]: hmm, the manpage still lists https://launchpad.net/pastebinit as homepage - I guess either are as good as the other.

**Review by raboof — APPROVED** [review]

**Pull request merged**